include/Dialect/PolyExt/IR/PolyExtDialect.td

```diff
@@ -0,0 +1,25 @@
+#ifndef HEIR_INCLUDE_DIALECT_POLYEXT_IR_POLYEXTDIALECT_TD_
+#define HEIR_INCLUDE_DIALECT_POLYEXT_IR_POLYEXTDIALECT_TD_
+
+include "mlir/IR/DialectBase.td"
+include "mlir/IR/AttrTypeBase.td"
+include "mlir/Interfaces/InferTypeOpInterface.td"
+
+//===----------------------------------------------------------------------===//
+// PolyExt dialect definitions
+//===----------------------------------------------------------------------===//
+
+def PolyExt_Dialect : Dialect {
+  // The namespace of the dialect.
+  // This corresponds to the string provided in `getDialectNamespace`.
+  let name = "poly_ext";
+
+  let description = [{
+    The PolyExt dialect is a place for "external" operations on
```

Comment:
nit: possibly add some more description here -are external operations that operate on the polynomial ring rather than between elements of the same polynomial ring?

I'm just drive-by reviewing right now, so I think I may be missing some context about why this is a separate dialect, and ops within the poly dialect can't transform the types attribute. i will find some time to catch up.

Reply:
I think the main reasons to have a separate dialect are (1) separate the crypto-specific ops from the parts of poly we would be realistically able to upstream (would have use beyond HEIR) and (2) because the ops here are not 
 ring operations (morphisms) in any of the usual senses, and hence fall under the category of "external" https://en.wikipedia.org/wiki/External_(mathematics).

Reply:
Ah, that makes a lot of sense. I like the idea anyway of keeping `poly` as neat as possible. Thanks!